prog/main.go

```diff
@@ -138,6 +142,41 @@ type appFlags struct {
 	consulInf       string
 }
 
+type containerLabelFiltersFlag struct {
+	apiTopologyOptions []app.APITopologyOption
+	exclude            bool
+}
+
+func (c *containerLabelFiltersFlag) String() string {
+	return fmt.Sprint(c.apiTopologyOptions)
+}
+
+func (c *containerLabelFiltersFlag) Set(flagValue string) error {
+	newAPITopologyOption, err := c.toAPITopologyOption(flagValue)
+	c.apiTopologyOptions = append(c.apiTopologyOptions, newAPITopologyOption)
+	return err
+}
+
+func (c *containerLabelFiltersFlag) toAPITopologyOption(flagValue string) (app.APITopologyOption, error) {
+	var indices []int
+	if indices = colonFinder.FindStringIndex(flagValue); len(indices) == 0 {
+		return app.APITopologyOption{}, fmt.Errorf("No unescaped colon found. This is needed to separate the title from the label")
+	} else if len(colonFinder.FindAllStringIndex(flagValue, -1)) > 1 {
+		return app.APITopologyOption{}, fmt.Errorf("Multiple unescaped colons. Escape colons that are part of the title and label")
+	}
+	var labelKeyValuePair []string
+	containerFilterTitle, containerFilterLabel := unescapeBackslashes.ReplaceAllString(flagValue[0:indices[0]+1], `$1`), unescapeBackslashes.ReplaceAllString(flagValue[indices[1]:], `$1`)
```

Comment:
Escaping before will require readjusting the colon separator index which may make things a bit complicated. Your call: give it a try a see how it looks.

Regardless, please try to create smaller lines, breaking the process in steps/intermediate variables if needed.